addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -183,7 +183,7 @@ def _timesheet_postprocess_values(self, values):
                 cost = timesheet.employee_id.timesheet_cost or 0.0
                 amount = -timesheet.unit_amount * cost
                 amount_converted = timesheet.employee_id.currency_id._convert(
-                    amount, timesheet.account_id.currency_id, self.env.company, timesheet.date)
+                    amount, timesheet.currency_id, self.env.company, timesheet.date)
```

Comment:
Hello @adrienpeiffer ,

Thanks for your fix 🙂 

Couldn't we rather do:
```suggestion
                    amount, timesheet.account_id.currency_id or timesheet.currency_id, self.env.company, timesheet.date)
```
In order to ensure that we modify the current flow the least we can?

May I also ask you to reformat the commit message as bellow?
```txt
As the company is not mandatory on both the employee and the analytical account,
if neither is filled in, it causes the application to crash.

This commit ensures that the currency of the current analytical line's company
is provided when no company is set on the analytic account.
```